### PR TITLE
Updates BYOND links to use HTTPS

### PR DESCRIPTION
**Pull request by ShizCalev**
Also makes a couple in-chat links use hrefs instead.

### CODE REVIEW

**Comment by DaxDupont** [comment]
Wasn't there a reason why we didn't do this before? I remember seeing this PR before.

**Comment by yourdoom9898** [comment]
@DaxDupont Maybe you're thinking of #33918 ?

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -248,7 +248,7 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 			log_access("Failed login: [key] - blacklisted byond version")
 			to_chat(src, "<span class='userdanger'>Your version of byond is blacklisted.</span>")
 			to_chat(src, "<span class='danger'>Byond build [byond_build] ([byond_version].[byond_build]) has been blacklisted for the following reason: [GLOB.blacklisted_builds[num2text(byond_build)]].</span>")
-			to_chat(src, "<span class='danger'>Please download a new version of byond. if [byond_build] is the latest, you can go to http://www.byond.com/download/build/ to download other versions.</span>")
+			to_chat(src, "<span class='danger'>Please download a new version of byond. if [byond_build] is the latest, you can go to <a href=\"https://secure.byond.com/download/build\">BYOND's website</a> to download other versions.</span>")
```
Comment (SpaceManiac) [inline code/modules/client/client_procs.dm]: `if` -> `If` while you're here

File: code/modules/client/client_procs.dm [hunk 2]
```diff
@@ -514,7 +514,7 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 	. = player_age
 
 /client/proc/findJoinDate()
-	var/list/http = world.Export("http://byond.com/members/[ckey]?format=text")
+	var/list/http = world.Export("https://secure.byond.com/members/[ckey]?format=text")
```
Comment (SpaceManiac) [inline code/modules/client/client_procs.dm]: Just tested, `world.Export` doesn't support HTTPS.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

**Review by SpaceManiac — DISMISSED** [review]

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**